src/main/java/net/glowstone/block/blocktype/BlockRedstoneTorch.java

```diff
@@ -55,7 +55,7 @@ public void placeBlock(GlowPlayer player, GlowBlockState state, BlockFace face,
     @Override
     public void updatePhysics(GlowBlock me) {
```

Comment:
Seems to be a problem with concurrency; this is the full stacktrace:

```
17:56:11 [SEVERE] Error occurred while pulsing world world
java.lang.ArrayIndexOutOfBoundsException: 35
        at java.util.LinkedList.toArray(Unknown Source)
        at java.util.ArrayList.<init>(Unknown Source)
        at net.glowstone.entity.GlowPlayer.processBlockChanges(GlowPlayer.java:619)
        at net.glowstone.entity.GlowPlayer.pulse(GlowPlayer.java:568)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
        at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.util.LinkedList$LLSpliterator.forEachRemaining(Unknown Source)
        at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
        at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.util.stream.ReferencePipeline.forEach(Unknown Source)
        at net.glowstone.GlowWorld.pulsePlayers(GlowWorld.java:483)
        at net.glowstone.GlowWorld.pulse(GlowWorld.java:364)
        at net.glowstone.scheduler.WorldScheduler$WorldThread.run(WorldScheduler.java:139)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
```
Something tells me that `blockChanges` list is out of sync of the tick map loop.